src/compiler.rs

```diff
@@ -39,6 +39,37 @@ macro_rules! expect {
     };
 }
 
+macro_rules! parse_branch {
+    ($compiler:expr, $block:expr, [ $( $call:expr ),* ]) => {
+        let block_length = $block.ops.len();
+        let token_length = $compiler.curr;
+        let num_errors = $compiler.errors.len();
+        let mut stored_errors = Vec::new();
+        let mut success = false;
+        // We risk getting a lot of errors if we are in an invalid state
+        // when we start the parse.
+        while !$compiler.panic {
+            $(
+                $call;
+                if !$compiler.panic {
+                    success = true;
+                    break;
+                }
+                $compiler.panic = false;
+                $compiler.curr = token_length;
+                let thrown_errors = $compiler.errors.len() - num_errors - 1;
+                stored_errors.extend($compiler.errors.split_off(thrown_errors));
+                $block.ops.truncate(block_length);
+            )*
+            break;
+        }
```

Comment:
Why have a `while` if we always break when all `$call`s have been made?

Reply:
Because I needed a block to break out from. But didn't want to introduce a function in a macro. I just used my C-skills.

Reply:
Ah, I see.